docs/site/Binding.md

```diff
@@ -0,0 +1,247 @@
+connects its value to a unique key as the address to access the entry in a
+context.
+
+### Attributes of a binding
+
+A binding typically has the following attributes:
+
+- key: Each binding has a `key` to uniquely identify itself within the context
+- scope: The scope controls how the binding value is created and cached within
+  the context
+- tags: Tags are names or name/value pairs to describe or annotate a binding
+- value: Each binding must be configured with a type of value provider so that
+  it be resolved to a constant or calculated value
+
+## How to create a binding?
+
+There are a few ways to create a binding:
+
+- Use `Binding` constructor:
+
+  ```ts
+  const binding = new Binding('my-key');
+  ```
+
+- Use `Binding.bind()`
+
+  ```ts
+  const binding = Binding.bind('my-key');
+  ```
+
+- Use `context.bind()`
+
+  ```ts
+  const context = new Context();
+  context.bind('my-key');
+  ```
+
+## How to set up a binding?
+
+The `Binding` class provides a set of fluent APIs to create and configure a
+binding.
+
+### Supply the value or a way to resolve the value
+
+The value can be supplied in one the following forms:
+
+#### A constant
+
+If binding is always resolved to a fixed value, we can bind it to a constant,
+which can be a string, a function, an object, an array, or any other types.
+
+```ts
+binding.to('my-value');
+```
+
+Please note the constant value cannot be a `Promise` to avoid confusions.
+
+#### A factory function
+
+Sometimes the value needs to be dynamically calculated, such as the current time
+or a value fetched from a remote service or database.
+
+```ts
+binding.toDynamicValue(() => 'my-value');
+binding.toDynamicValue(() => new Date());
+binding.toDynamicValue(() => Promise.resolve('my-value'));
+```
+
+#### A class
+
+The binding can represent an instance of a class, for example, a controller. A
+class can be used to instantiate an instance as the resolved value. Dependency
+injection is often leveraged for its members.
+
+```ts
+class MyController {
+  constructor(@inject('my-options') private options: MyOptions) {
+    // ...
+  }
+}
+
+binding.toClass(MyController);
+```
+
+#### A provider
+
+A provider is a class with `value()` method to calculate the value from its
+instance. The main reason to use a provider class is to leverage dependency
+injection for the factory function.
+
+```ts
+class MyValueProvider implements Provider<string> {
+  constructor(@inject('my-options') private options: MyOptions) {
+    // ...
+  }
+
+  value() {
+    return this.options.defaultValue;
+  }
+}
+
+binding.toProvider(MyValueProvider);
+```
+
+### Configure the scope
+
+We allow a binding to be resolved within a context using one of the following
+scopes:
+
+- BindingScope.TRANSIENT (default)
+- BindingScope.CONTEXT
+- BindingScope.SINGLETON
+
+For a complete list of descriptions, please see
+[BindingScope](http://apidocs.loopback.io/@loopback%2fdocs/context.html#BindingScope).
```

Comment:
The documentation rendered for BindingScope has formatting that looks a bit weird to me. Would you like to take a look and perhaps fix it as part of this PR?

<img width="644" alt="screen shot 2018-12-06 at 09 47 20" src="https://user-images.githubusercontent.com/1140553/49572380-f1357800-f93b-11e8-8dcb-19cc48437e24.png">